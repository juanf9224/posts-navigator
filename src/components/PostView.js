import React, { Fragment, useState } from "react";

import PostEdit from './PostEdit';
import PostDetail from './PostDetail';

const PostView = () => {
  const [isEdit, setIsEdit] = useState(false);

  return (
    <Fragment>
      {isEdit ? (
        <PostEdit setIsEdit={setIsEdit} />
      ) : (
        <PostDetail setIsEdit={setIsEdit} />
      )}
    </Fragment>
  );
};

export { PostView as default };
