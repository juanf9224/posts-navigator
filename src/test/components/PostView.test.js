import React from "react";
import { shallow } from "enzyme";

import PostView from "../../components/PostView";
import AutocompleteContext from "../../context/autocomplete";

describe("PostView component test suite", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <AutocompleteContext.Provider>
        <PostView />
      </AutocompleteContext.Provider>
    );
  });

  test("it should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
