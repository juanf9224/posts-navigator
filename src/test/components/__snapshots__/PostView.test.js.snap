// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostView component test suite it should match snapshot 1`] = `ShallowWrapper {}`;
